deployment/environment/memory/chain.go

```diff
@@ -86,6 +88,41 @@ func getTestSolanaChainSelectors() []uint64 {
 	return result
 }
 
+func generateSolanaKeypair(t testing.TB) (solana.PrivateKey, string, error) {
+	// Create a temporary directory that will be cleaned up after the test
+	tmpDir := t.TempDir()
+
+	privateKey, err := solana.NewRandomPrivateKey()
+	if err != nil {
+		return solana.PrivateKey{}, "", fmt.Errorf("failed to generate private key: %w", err)
+	}
+
+	// Convert private key bytes to JSON array
+	privateKeyBytes, err := base58.Decode(privateKey.String())
+	if err != nil {
+		return solana.PrivateKey{}, "", fmt.Errorf("failed to decode private key: %w", err)
+	}
```

Comment:
Just cast the type, it's already `[]byte`

```suggestion
	privateKeyBytes := []byte(privateKey)
```

Reply:
Will handle this in the home chain PR